src/graph/CreateTagExecutor.cpp

```diff
@@ -18,25 +18,97 @@ CreateTagExecutor::CreateTagExecutor(Sentence *sentence,
 
 
 Status CreateTagExecutor::prepare() {
-    return checkIfGraphSpaceChosen();
-}
+    auto status = checkIfGraphSpaceChosen();
 
+    if (!status.ok()) {
+        return status;
+    }
 
-void CreateTagExecutor::execute() {
-    auto *mc = ectx()->getMetaClient();
-    auto *name = sentence_->name();
     const auto& specs = sentence_->columnSpecs();
-    auto spaceId = ectx()->rctx()->session()->space();
+    const auto& schemaProps = sentence_->getSchemaProps();
 
-    nebula::cpp2::Schema schema;
     for (auto& spec : specs) {
         nebula::cpp2::ColumnDef column;
         column.name = *spec->name();
         column.type.type = columnTypeToSupportedType(spec->type());
-        schema.columns.emplace_back(std::move(column));
+        schema_.columns.emplace_back(std::move(column));
     }
 
-    auto future = mc->createTagSchema(spaceId, *name, schema);
+    // set schema_.schema_prop default
+    // 0 means infinity
+    schema_.schema_prop.set_ttl_duration(0);
```

Comment:
It can init in the constructor of schema_prop

Reply:
👍👍👍，good  catch！